stingray/lightcurve.py

```diff
@@ -175,3 +209,163 @@ def rebin_lightcurve(self, dt_new, method='sum'):
 
         lc_new = Lightcurve(bin_time, bin_counts)
         return lc_new
+
+    def join(self, other):
+        """
+        Join two lightcurves into a single object.
+
+        The new Lightcurve object will contain time stamps from both the
+        objects. The individual count per bin in the resulting object will
+        be the sum total of the count per bin in both lightcurve objects.
+
+        Parameters
+        ----------
+        other : Lightcurve object
+            The other Lightcurve object which is supposed to be joined with.
+
+        Returns
+        -------
+        lc_new : Lightcurve object
+            The resulting lightcurve object.
+
+        Example
+        -------
+        >>> time1 = [5, 10, 15]
+        >>> count1 = [300, 100, 400]
+        >>> time2 = [15, 20, 25]
+        >>> count2 = [600, 1200, 800]
+        >>> lc1 = Lightcurve(time1, count1)
+        >>> lc2 = Lightcurve(time2, count2)
+        >>> lc = lc1.join(lc2)
+        >>> lc.time
+        array([ 5, 10, 15, 20, 25])
+        >>> lc.counts
+        array([ 300,  100, 1000, 1200,  800])
+        """
+        assert self.dt == other.dt, "The bin width of both the lightcurves" \
+                                    " must be same."
+
```

Comment:
I think analogously to #78, this should probably throw a warning rather than an error. There's no intrinsic reason why the bin widths of the two time series must be the same, except that it'll screw up some of the Fourier methods. However, we might include methods in the future where this doesn't matter, and it might still be useful, for example, for visualization on the GUI.